pkg/csi/csi.go

```diff
@@ -58,19 +59,37 @@ func InitializeSidecars(ver string) error {
 		return err
 	}
 
-	sv := version.MustParseGeneric(ver)
-
-	for _, image := range si {
-		v := version.MustParseGeneric(image.Version)
-		if sv.Major() == v.Major() && sv.Minor() == v.Minor() {
-			sidecarImages = &image
-			break
+	pltVersion := version.MustParseGeneric(ver)
+
+	// this helps in not parsing the version twice in loop.
+	closestVersion := pltVersion.WithMinor(0)
+	for idx := range si {
+		siVersion := version.MustParseGeneric(si[idx].Version)
+		log.Info("verifying sidecar images version for best compatibility to platform",
+			"sidecar", siVersion, "platform", pltVersion)
```

Comment:
1. This is not the way we do multiline statement in this codebase
2. You are not verifying here, you are checking
3. Cosntant parts should be part of the message itself

```suggestion
		log.Info(
			"Searching for the most compatible CSI image version: CSI: %s, Platform: %s",
			siVersion, 
			pltVersion,
		)
```

Reply:
ack for the rest
> 3. Cosntant parts should be part of the message itself
- logr package support key & value pairs to be part of log, I'll need to use fmt.Sprintf to take the change